homeassistant/components/starlink/entity.py

```diff
@@ -0,0 +1,64 @@
+"""Contains base entity classes for Starlink entities."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import datetime
+
+from starlink_grpc import StatusDict
+
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import StarlinkUpdateCoordinator
+
+
+@dataclass
+class StarlinkSensorEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[StatusDict], datetime | StateType]
+
+
+@dataclass
+class StarlinkSensorEntityDescription(
+    SensorEntityDescription, StarlinkSensorEntityDescriptionMixin
+):
+    """Describes a Starlink sensor entity."""
+
+
+class StarlinkSensorEntity(CoordinatorEntity[StarlinkUpdateCoordinator], SensorEntity):
```

Comment:
I missed this on the first review but commented here: https://github.com/home-assistant/core/pull/85409#discussion_r1064096540

This should be split up into a `StarlinkSensorEntity` and a base class `StarlinkEntity` for the entity (the non-sensor) parts should live in `entity.py` and the sensor specific parts should move to `sensor.py`